pkg/agent/openflow/network_policy.go

```diff
@@ -1823,14 +1827,72 @@ func newFeatureNetworkPolicy(
 }
 
 func (f *featureNetworkPolicy) initFlows() []binding.Flow {
-	f.egressTables = map[uint8]struct{}{EgressRuleTable.GetID(): {}, EgressDefaultTable.GetID(): {}}
+	f.egressTables = map[uint8]struct{}{EgressDefaultTable.GetID(): {}}
 	if f.enableAntreaPolicy {
 		f.egressTables[AntreaPolicyEgressRuleTable.GetID()] = struct{}{}
 	}
 	var flows []binding.Flow
-	flows = append(flows, f.establishedConnectionFlows()...)
-	flows = append(flows, f.relatedConnectionFlows()...)
-	flows = append(flows, f.rejectBypassNetworkpolicyFlows()...)
-	flows = append(flows, f.ingressClassifierFlows()...)
+	if f.nodeType == config.K8sNode {
+		f.egressTables[EgressRuleTable.GetID()] = struct{}{}
+		flows = append(flows, f.establishedConnectionFlows()...)
+		flows = append(flows, f.relatedConnectionFlows()...)
+		flows = append(flows, f.rejectBypassNetworkpolicyFlows()...)
+		flows = append(flows, f.ingressClassifierFlows()...)
+	}
+	if f.nodeType == config.ExternalNode {
+		flows = append(flows, f.bypassPolicyRuleCheckFlows()...)
+	}
+	return flows
+}
+
+// bypassPolicyRuleCheckFlows installs OpenFlow entries to resubmit the packets in an established or related
+// connections to the metric table in the same stage directly. It also resubmits the packets marked with
+// "CustomReasonReject" to the metric table. So that these packets would skip the check on NetworkPolicy rules.
+func (f *featureNetworkPolicy) bypassPolicyRuleCheckFlows() []binding.Flow {
+	var flows []binding.Flow
+	for _, ipProtocol := range f.ipProtocols {
+		flows = append(flows,
+			EgressSecurityClassifierTable.ofTable.BuildFlow(priorityHigh).
```

Comment:
Are these flows overlap with establishedConnectionFlows, relatedConnectionFlows, rejectBypassNetworkpolicyFlows, etc.? Could we share code?

Reply:
Actually, the flows in funciton establishedConnectionFlows, relatedConnectionFlows, and rejectBypassNetworkpolicyFlows are similar to this function. The difference is the flows in container case are installed in all of these rule tables with high priority: EgressRuleTable/IngressRuleTable, AntreaPolicyEgressRuleTable/AntreaPolicyIngressRuleTable, EgressDefaultTable/IngressDefaultTable. In think this is a bit duplicated. So in VM Agent case, I only install these flows in the first table of SecurityEgressStage/SecurityIngressStage, and if the packet is a established/related/replied in an existing connection, it jumps to the related metric table directly.

I think the code in this change is more clean, but I would like to hear your points to see if we should use it in container case. @jianjuns @Dyanngg @tnqn

Reply:
It is always better to share one design. @Dyanngg : comments?

Reply:
@wenyingd Are you suggesting that for container case, we switch to installing these flows in the first table of SecurityEgressStage/SecurityIngressStage as well?

Reply:
> @wenyingd Are you suggesting that for container case, we switch to installing these flows in the first table of SecurityEgressStage/SecurityIngressStage as well? @GraysonWu for awareness as he implemented rejectBypassNetworkpolicyFlows

Yes.

Reply:
Actually, `rejectBypassNetowrkpolicyFlows` has been removed in PR #3686. Since we changed our reject packet_out logic they are not in use anymore.
Putting all those "skip/bypass" flows together sounds clean to me as long as it won't cause any side effects.

Reply:
I tried to replace the existing code with the new format that bypass established/replied packets from the first table in Egress/IngressSecurityStage. Then I found the e2e test case `ACNPFQDNPolicy` is failed. After checking the code, I found that the DNS conjunction rule is adding in table `AntreaPolicyIngressRuleTable`, and it sends the DNS response packet ( +rpl ) packet to Antrea Agent, or sends the reject packet-out message to Output table. With the new design, the DNS response packet would not enters `AntreaPolicyIngressRuleTable` but goes into IngressMetricTable directly, which leads to failure. So I want to change the design as: If ANP is enabled, the flows are installed in `AntreaPolicyEgressRuleTable/AntreaPolicyIngressRuleTable`, otherwise, they are installed in `EgressRuleTable/IngressRuleTable`. Anyway, I want to install these flows only one time. What do you think? @GraysonWu @Dyanngg @jianjuns

Reply:
> or sends the reject packet-out message to Output table.

I think you mean DNS response packet-out message here.

If this is the case, when ANP is enabled in AntreaPolicyIngressRuleTable the priority order should be:
```
DNS resp packet-out msg bypass flow > DNS resp intercept flow > other +rpl bypass flows(if there's any) > policy flows
```
Is that right?
Let's see @Dyanngg 's ideas.